homeassistant/components/here_weather/sensor.py

```diff
@@ -0,0 +1,166 @@
+"""Support for the HERE Destination Weather service."""
+import logging
+from typing import Callable, Dict, Optional, Union
+
+import herepy
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (
+    CONF_LATITUDE,
+    CONF_LONGITUDE,
+    CONF_MODE,
+    CONF_NAME,
+    CONF_UNIT_SYSTEM,
+    CONF_UNIT_SYSTEM_IMPERIAL,
+    CONF_UNIT_SYSTEM_METRIC,
+)
+from homeassistant.core import HomeAssistant
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+
+from . import (
+    HEREWeatherData,
+    convert_unit_of_measurement_if_needed,
+    get_attribute_from_here_data,
+)
+from .const import (
+    CONF_API_KEY,
+    CONF_LOCATION_NAME,
+    CONF_MODES,
+    CONF_OFFSET,
+    CONF_ZIP_CODE,
+    DEFAULT_MODE,
+    DEFAULT_NAME,
+    SENSOR_TYPES,
+)
+
+UNITS = [CONF_UNIT_SYSTEM_METRIC, CONF_UNIT_SYSTEM_IMPERIAL]
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
```

Comment:
We no longer accept additions or changes to the yaml config of integrations. See: https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md

This integration needs to be refactored to use a config flow and config entries.

https://developers.home-assistant.io/docs/config_entries_config_flow_handler

Reply:
Took me some time to,...well, find time.^^ I implemented the config flow. Is there a better option for the values of the dropdown lists for `CONF_MODE` and `CONF_OPTION`. Currently they are hardcoded and cannot be translated.